feat: deterministic / reproducible build support (#1641)
goreleaser/goreleaser: www/docs/customization/build.md

GoReleaser uses `git describe` to get the build tag. You can set
a different build tag using the environment variable `GORELEASER_CURRENT_TAG`.
This is useful in scenarios where two tags point to the same commit.

## Reproducible Builds

To make your releases, checksums, and signatures reproducible, you will need to make some (if not all) of the following modifications to the build defaults in GoReleaser:

* Modify `ldflags`: by default `main.Date` is set to the time GoReleaser is run (`{{.Date}}`), you can set this to `{{.CommitDate}}` or just not pass the variable.

> * Modify `mod_timestamp`: by default this is set to the timestamp when GoReleaser is run (`{{.Timestamp}}`), set to `{{.CommitTimestamp}}` or a constant value instead.

Author: ```suggestion
* Modify `mod_timestamp`: by default this is empty string, set to `{{.CommitTimestamp}}` or a constant value instead.
```

* If you do not run your builds from a consistent directory structure, pass `-trimpath` to `flags`.

> * Remove uses of the `time` template function that could. This function returns a new value on every call and is not deterministic.

Author: ```suggestion
* Remove uses of the `time` template function. This function returns a new value on every call and is not deterministic.
```
